### DISCUSSION TITLE
Text On Objects?

### DISCUSSION THREAD

**Discussion by aminmix** [KilledByAPixel/LittleJS#71] [Q&A] [answered]
Hello , thank you for this awesome project <3
I am new to LittleJS and I need to put static text on my bricks made by engineObject
for example ; 
```
class Brick extends EngineObject {
    constructor(pos){
super(pos,vec2(1.25,1));
this.innerText="Hello"   // It doesnt work!
    }
```
Is there any way to put text in objects?

**Comment by KilledByAPixel** [comment, marked as answer]
Hey, yes that is super easy to do! You can override the default render and use the drawText function...

```
render()
{
    super.render();
    drawText(this.innerText, this.pos, this.size);
}
```

**Reply by aminmix** [reply]
Thanks , It worked like a charm but how can i set the color of my text ? Its white by default

**Comment by KilledByAPixel** [comment]
Yeah, there are a few ways you can find out...

If you mouse over the function in vs it will show you the params. Color is the 4th param.

![2024-02-09 18_43_02-● gameObjects js - LittleJS - Visual Studio Code](https://github.com/KilledByAPixel/LittleJS/assets/16227600/bc316f55-0f71-4148-a2c9-89f3e868526b)

You can also check the docs...

[https://killedbyapixel.github.io/LittleJS/docs/Draw.html#.drawText](url)

Or the function in the code...

[https://github.com/KilledByAPixel/LittleJS/blob/main/src/engineDraw.js#L250](url)

**Reply by aminmix** [reply]
Great @};-
